thunder/core/transforms.py

```diff
@@ -1485,7 +1493,7 @@ def symbol_to_eval(bound_symbol):
 )
 
 
-def eval_trace(trace, *args, symbol_mapper=symbol_to_eval, with_env=False, **kwargs):
+def eval_trace(trace, *args, symbol_mapper=symbol_to_eval, with_env=False, allow_duplicates=False, **kwargs):
```

Comment:
```suggestion
def eval_trace(trace, *args, symbol_mapper=symbol_to_eval, with_env=False, **kwargs):
```
to not need `allow_duplicates` here. (And we don't need the changes below.)

Reply:
However, when we do want to allow duplicates (for other reasons) won't it be better to keep these changes?

Reply:
So we don't allow the duplicates because it would be problematic, so I would venture we don't add the override option until we really know we need it.

Reply:
Sounds great! Let me update it :)